docs/data-layer.md

```diff
@@ -0,0 +1,178 @@
+
+``` javascript
+connect(({ eholdings: { data } }) => ({
+  // ...mapStateToProps
+  // data needed by the component is accessed here
+}), {
+  // ...mapDispatchToProps
+  // actions used to send requests for data are defined here
+});
+```
+
+When data is needed, you must first request it with one of the actions
+defined in `mapDispatchToProps`. The data received in
+`mapStateToProps` will then update with a pending request, and update
+again once the request resolves or rejects.
+
+## Models
+
+In eHoldings, you must define a model to request and store data in the
+redux store.
+
+``` javascript
+// src/redux/provider.js
+import model, { hasMany } from './model';
+
+class Provider {
+  // if an attribute is not defined here, the data will not be accessible in our component
+  name = '';
+  packagesSelected = 0;
+  packagesTotal = 0;
+
+  // relationships are defined using `hasMany(type)` or `belongsTo(type)`
+  // the type defaults to the relationship key
+  packages = hasMany();
+}
+
+export default model({
+  type: 'providers',           // used as the key when storing data
+  path: '/eholdings/providers' // used in the URL during requests
+})(Provider);
+```
+
+### Requesting & Resolving Data
+
+Using this model and the eHoldings resolver within `connect` allows us
+to request and receive data from the backend Okapi server.
+
+``` javascript
+// src/routes/provider-show.js
+import { createResolver } from '../redux';
+import Provider from '../redux/provider';
+
+// ... class ProviderShowRoute extends Component {}
+
+export default connect(
+  ({ eholdings: { data } }, { match }) => {
+    let resolver = createResolver(data);
+
+    return {
+      // returns a `Provider` model we defined earlier
+      model: resolver.find('providers', match.params.providerId),
+      // used in the component to look up packages
+      resolver
+    };
+  }, {
+    // will request a provider at `/eholdings/provider/:id`
+    getProvider: id => Provider.find(id, { include: 'packages' }),
+    // will request packages at `/eholdings/provider/:id/packages?:params`
+    getPackages: (id, params) => Provider.queryRelated(id, 'packages', params)
+  }
+)(ProviderShowRoute);
+```
+
+The model returned by the resolver has request objects at
+`model.request`, `model.update`, and `model.destroy` which provide
+information about the different types of requests, such as
+`isPending`, `isResolved`, `isRejected`, `errors`, etc.
+
+When dealing with a collection of models, such as with `hasMany`, the
+returned collection object lazily looks up other pages from the same
+query in the data store. This allows you to map over all records that
+have been returned by a query regardless of page, such as when
+rendering the search results lists.
+
+For search results, the resolver must know the exact params you passed
+to the search action in order to properly find the records from the
+request. In the case of provider packages, the search params are
```

Comment:
is `provider package` synonymous with `managed package`?  don't think i've heard this term used much

Reply:
`provider packages` as in packages belonging to a provider

Reply:
Maybe I should change this to `In the case of a provider's packages`

Reply:
i think it becomes pretty clear further down but it couldn't hurt 🤷‍♀️